hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -181,7 +181,8 @@ private ClientMessage handleAuthenticated() {
         Connection connection = endpoint.getConnection();
         ILogger logger = clientEngine.getLogger(getClass());
 
-        endpoint.authenticated(principal, credentials, isOwnerConnection(), clientVersion);
+        clientEngine.trySetLastAuthenticationCorrelationId(principal.getUuid(), clientMessage.getCorrelationId());
```

Comment:
Should we check the boolean result of this call and throw exception? (similar as done at master)